src/core/exec-invoke.c

```diff
@@ -4904,7 +4904,7 @@ static void prepare_terminal(
 
         (void) exec_context_apply_tty_size(context, STDIN_FILENO, STDOUT_FILENO, /* tty_path= */ NULL);
 
-        if (use_ansi)
+        if (use_ansi && !getenv_terminal_is_dumb())
```

Comment:
this should not be necessary: exec_context_shall_ansi_seq_reset() already contains a suitable check for dumb terminals.